dao/src/main/java/org/thingsboard/server/dao/rule/BaseRuleChainService.java

```diff
@@ -748,6 +748,7 @@ private void deleteRuleNodes(TenantId tenantId, List<RuleNode> ruleNodes) {
     }
 
     @Override
+    @Transactional
```

Comment:
BTW. check who is calling this method. there is a single call that is already transactional. But as it is already public, you can put transactional over here. Please, add the test